### PR TITLE
std_sbom_siemens_filename_direct_to_dev

### PR DESCRIPTION
**Pull request by Vijayalakshmi027**
net  8 migration changes
SBOM property addition direct and filename
UT and IT fixes
Docker file update to .net 8
Github compile yml update to .net 8

UT and IT passes in the dev environment.

### CODE REVIEW

**Comment by Vijayalakshmi027** [comment]
All UT and IT passed locally
image

File: src/SW360IntegrationTest/NPM/PackageIdentifierInitialTestMode.cs [hunk 1]
```diff
@@ -34,6 +46,50 @@ public void Setup()
...
+            using var httpClient = new HttpClient();
+            httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(testParameters.SW360AuthTokenType, testParameters.SW360AuthTokenValue);
+            string componentName = "samplecomponent";
+            string componentVersion = "1.0.0";
+            string componentType = "OSS";
+            HttpResponseMessage componentCheck = await httpClient.GetAsync(TestConstant.Sw360ReleaseApi);
+
+            // Act
+            //string componentCheckText = await componentCheck.Content.ReadAsStringAsync();
```
Comment (karthika-g) [inline src/SW360IntegrationTest/NPM/PackageIdentifierInitialTestMode.cs:61]: remove commented code

File: src/LCT.PackageIdentifier/Program.cs [hunk 2]
```diff
@@ -178,6 +178,12 @@ private static string DisplayInclude(CommonAppSettings appSettings)
                         totalString = string.Join(",", appSettings.Conan.Include?.ToList());
                     }
                     return totalString;
+                case "ALPINE":
+                    if (appSettings.Npm.Include != null)
+                    {
+                        totalString = string.Join(",", appSettings.Npm.Include?.ToList());
```
Comment (karthika-g) [inline src/LCT.PackageIdentifier/Program.cs:184]: why is npm include section checked for alpine type?

File: src/LCT.PackageIdentifier/Program.cs [hunk 3]
```diff
@@ -226,6 +232,12 @@ private static string DisplayExclude(CommonAppSettings appSettings)
                         totalString = string.Join(",", appSettings.Conan.Exclude?.ToList());
                     }
                     return totalString;
+                case "ALPINE":
+                    if (appSettings.Npm.Include != null)
+                    {
```
Comment (karthika-g) [inline src/LCT.PackageIdentifier/Program.cs:237]: why is npm include section checked for alpine type

File: src/LCT.PackageIdentifier/Program.cs [hunk 4]
```diff
@@ -276,6 +288,12 @@ private static string DisplayExcludeComponents(CommonAppSettings appSettings)
                         totalString = string.Join(",", appSettings.Conan.ExcludedComponents?.ToList());
                     }
                     return totalString;
+                case "ALPINE":
+                    if (appSettings.Npm.Include != null)
```
Comment (karthika-g) [inline src/LCT.PackageIdentifier/Program.cs:292]: why is npm include section checked for alpine type

File: src/LCT.PackageIdentifier/NugetDevDependencyParser.cs [hunk 5]
```diff
@@ -174,6 +179,41 @@ private static void ParseJsonFile(string filePath, Container container)
             }
         }
 
+        private static void GetDirectDependencies(string filePath)
+        {
+            var readValue = File.ReadAllText(filePath);
+            JObject serializedContent = JObject.Parse(readValue);
+            JToken projectFramoworks = serializedContent["project"]["frameworks"];
+            if (projectFramoworks == null && !projectFramoworks.HasValues)
```
Comment (karthika-g) [inline src/LCT.PackageIdentifier/NugetDevDependencyParser.cs:187]: projectFrameworks

File: src/LCT.PackageIdentifier/NpmProcessor.cs [hunk 6]
```diff
@@ -181,8 +181,14 @@ private static void CreateFileForMultipleVersions(List<Component> componentsWith
         private static void GetPackagesForBom(string filepath, ref List<BundledComponents> bundledComponents, ref List<Component> lstComponentForBOM, ref int noOfDevDependent, IEnumerable<JProperty> depencyComponentList)
         {
             BomCreator.bomKpiData.ComponentsinPackageLockJsonFile += depencyComponentList.Count();
-
-            foreach (JProperty prop in depencyComponentList)
+            var prop2 = depencyComponentList.ToList()[0];
```
Comment (karthika-g) [inline src/LCT.PackageIdentifier/NpmProcessor.cs:184]: give meaning full names for variables

**Review by adityanarayanp — APPROVED** [review]

**Pull request merged**